tests/Unit/Remote/PayloadTest.php

```diff
@@ -26,8 +26,8 @@ class PayloadTest extends TestCase
 
     protected function setUp(): void
     {
-        $this->process = $this->prophesize(Process::class);
-        $this->processFactory = $this->prophesize(ProcessFactory::class);
+        $this->process = $this->createMock(Process::class);
+        $this->processFactory = $this->createMock(ProcessFactory::class);
     }
```

Comment:
instead of this you can:

```
use ProphecyTrait;
```

this is the approach going forward

Reply:
Ah, you are using phpspec and phpunit together. I was aiming to keep maintainability effort low by simplifying the test to use phpunit only.

Reply:
I think in general I'd like to use the native PHPUnit mock framework, but would rather do this all together rather than mixing the two approaches. Using prophecy and that trait means nothing needs to change. The PHPUnit approach doesn't look as bad as I remember though ,,,

Reply:
I'd still say, you don't want to mock `ProcessFactory` but instantiate it.

Reply:
> but would rather do this all together rather than mixing the two approaches
You are mixing two approaches currently (and one of them is deprecated).
I think it's more realistic to improve software over time, not at once.
It gives you the freedom to apply changes iteratively, just as you need them.

Reply:
> I'd still say, you don't want to mock ProcessFactory but instantiate it.

We need to mock the ProcessFactory (this is why it exists) there could indeed be an interface.

> You are mixing two approaches currently

No, we only use prophecy so far (AFAICS)